### ISSUE TITLE
Error: [$injector:unpr] - no simulations displayed

### ISSUE THREAD

**Issue by mrakitin** [radiasoft/sirepo#395]
With recent updates there is the following error in JS console log:
```js
Error: [$injector:unpr] http://errors.angularjs.org/1.4.2/$injector/unpr?p0=_Provider%20%3C-%20_%20%3C-%20appState%20%3C-%20appHeaderDirective
J/<@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:6:416
eb/r.$injector<@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:40:375
d@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:38:364
eb/x.$injector<@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:40:447
d@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:38:364
e@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:39:124
e/<@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:37:96
e@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:39:154
eb/x.$injector<@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:40:476
d@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:38:364
e@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:39:124
s/</<@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:50:55
m@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:7:320
s/<@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:50:24
e@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:39:154
e/<@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:37:96
e@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:39:154
eb/x.$injector<@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:40:476
d@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:38:364
A@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:66:349
ha@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:56:381
S@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:55:90
S@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:55:258
S@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:55:258
S@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:55:258
S@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:55:258
S@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:55:258
Z@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:53:85
Ac/d/</<@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:19:477
jf/this.$get</n.prototype.$eval@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:134:489
jf/this.$get</n.prototype.$apply@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:135:212
Ac/d/<@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:19:435
e@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:39:154
Ac/d@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:19:356
Ac@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:20:151
.success@http://nsls2expdev1:8000/static/js/sirepo.js?20160921.022530:21:13
n.Callbacks/j@http://nsls2expdev1:8000/static/js/ext/jquery-2.1.4.min.js?20160921.022530:2:26920
n.Callbacks/k.fireWith@http://nsls2expdev1:8000/static/js/ext/jquery-2.1.4.min.js?20160921.022530:2:27738
x@http://nsls2expdev1:8000/static/js/ext/jquery-2.1.4.min.js?20160921.022530:4:11251
.send/b/<@http://nsls2expdev1:8000/static/js/ext/jquery-2.1.4.min.js?20160921.022530:4:14765
```

In the server log:
```
10.0.2.2 - - [21/Sep/2016 02:25:34] "POST /error-logging HTTP/1.1" 200 -
../sirepo/server.py:192:app_error_logging 10.0.2.2: javascript error: {
    "cause": "<no cause>",
    "message": "Error: [$injector:unpr] http://errors.angularjs.org/1.4.2/$injector/unpr?p0=_Provider%20%3C-%20_%20%3C-%20appState%20%3C-%20activeSection",
    "stackTrace": [
        "J/<@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:6:416",
        "eb/r.$injector<@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:40:375",
        "d@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:38:364",
        "eb/x.$injector<@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:40:447",
        "d@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:38:364",
        "e@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:39:124",
        "e/<@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:37:96",
        "e@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:39:154",
        "eb/x.$injector<@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:40:476",
        "d@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:38:364",
        "e@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:39:124",
        "e/<@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:37:96",
        "e@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:39:154",
        "eb/x.$injector<@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:40:476",
        "d@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:38:364",
        "e@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:39:124",
        "Xe/this.$get</</<@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:80:208",
        "S@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:61:272",
        "g@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:54:492",
        "g@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:55:14",
        "Z/<@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:54:51",
        "Ac/d/</<@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:19:477",
        "jf/this.$get</n.prototype.$eval@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:134:489",
        "jf/this.$get</n.prototype.$apply@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:135:212",
        "Ac/d/<@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:19:435",
        "e@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:39:154",
        "Ac/d@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:19:356",
        "Ac@http://nsls2expdev1:8000/static/js/ext/angular-1.4.2.min.js?20160921.022530:20:151",
        ".success@http://nsls2expdev1:8000/static/js/sirepo.js?20160921.022530:21:13",
        "n.Callbacks/j@http://nsls2expdev1:8000/static/js/ext/jquery-2.1.4.min.js?20160921.022530:2:26920",
        "n.Callbacks/k.fireWith@http://nsls2expdev1:8000/static/js/ext/jquery-2.1.4.min.js?20160921.022530:2:27738",
        "x@http://nsls2expdev1:8000/static/js/ext/jquery-2.1.4.min.js?20160921.022530:4:11251",
        ".send/b/<@http://nsls2expdev1:8000/static/js/ext/jquery-2.1.4.min.js?20160921.022530:4:14765"
    ],
    "type": "exception",
    "url": "http://nsls2expdev1:8000/srw"
}
```

**Comment by robnagler** [comment]
Did you pull absolutely latest? https://github.com/radiasoft/sirepo/commit/90576fe30c688af675461286fad42ac4abf0b527

**Comment by mrakitin** [comment]
Yes, I'm on 90576fe30c688af675461286fad42ac4abf0b527 on master. When I checked out to mrakitin c619593e308a0b34641e6a31dba7abcc975c15f2, everything worked fine. I used Firefox and Chrome.

```bash
✔ ~/src/radiasoft/sirepo [master|✔]
02:54 $ git log -n 1
commit 90576fe30c688af675461286fad42ac4abf0b527
Author: robnagler <github@q33.us>
Date:   Wed Sep 21 01:23:49 2016 +0000

    fix https://travis-ci.org/radiasoft/sirepo/builds/161488830
✔ ~/src/radiasoft/sirepo [master|✔]
02:54 $ git checkout mrakitin
Switched to branch 'mrakitin'
Your branch is up-to-date with 'origin/mrakitin'.
✔ ~/src/radiasoft/sirepo [mrakitin|✔]
02:54 $ git log -n 1
commit c619593e308a0b34641e6a31dba7abcc975c15f2
Author: robnagler <github@q33.us>
Date:   Mon Sep 19 23:40:07 2016 +0000

    Fix #384: returning redirect as an exception instead of expecting the data to be interpreted by intermediate layers
✔ ~/src/radiasoft/sirepo [mrakitin|✔]
02:54 $
```

**Comment by mrakitin** [comment]
The same issue is in my dev environment.

**Comment by mrakitin** [comment]
According to https://solidfoundationwebdev.com/blog/posts/how-to-use-underscore-in-your-angularjs-controllers, `underscore` must be declared as a dependency of the module:
```js
// Declare it as a dependency of your module
var app = angular.module('app', ['underscore']);
```
I'll fix the issue.

**Comment by mrakitin** [comment]
Fixed in 96f564d.
